File: sdks/java/src/main/java/com/digows/whatsappgateway/activation/ActivationMode.java
package com.digows.whatsappgateway.activation;

import com.fasterxml.jackson.annotation.JsonCreator;
import com.fasterxml.jackson.annotation.JsonValue;

public enum ActivationMode
{
  QR_CODE("qr"),
  PAIRING_CODE("pairing_code");

  private final String wireValue;

  ActivationMode(String wireValue)
  {
    this.wireValue = wireValue;
  }

  @JsonValue
  public String getWireValue()
  {
    return wireValue;
  }

  @JsonCreator
  public static ActivationMode fromWireValue(String wireValue)
  {
    for (ActivationMode value : values())
    {
      if (value.wireValue.equals(wireValue))
      {
        return value;
      }
    }

    throw new IllegalArgumentException("Unsupported activation mode \"" + wireValue + "\".");
  }
}

File: runtime/src/domain/entities/activation/ActivationMode.ts
/**
 * Supported activation strategies for pairing a WhatsApp session.
 */
export enum ActivationMode {
  QrCode = 'qr',
  PairingCode = 'pairing_code',
}

export function parseActivationMode(value: string): ActivationMode {
  switch (value) {
    case ActivationMode.QrCode:
      return ActivationMode.QrCode;
    case ActivationMode.PairingCode:
      return ActivationMode.PairingCode;
    default:
      throw new Error(`Unsupported activation mode "${value}".`);
  }
}
